docs/fundamentals/code-analysis/quality-rules/ca2256.md

```diff
@@ -0,0 +1,40 @@
+---
+title: "CA2256: All members declared in parent interfaces must have an implementation in a DynamicInterfaceCastableImplementation-attributed interface"
+description: "Learn about code analysis rule CA2256: All members declared in parent interfaces must have an implementation in a DynamicInterfaceCastableImplementation-attributed interface"
+ms.date: 01/19/2022
+ms.topic: reference
+f1_keywords:
+- CA2256
+- DynamicInterfaceCastableImplementationAnalyzer
+helpviewer_keywords:
+- DynamicInterfaceCastableImplementationAnalyzer
+- CA2256
+author: Youssef1313
+---
+# CA2256: All members declared in parent interfaces must have an implementation in a DynamicInterfaceCastableImplementation-attributed interface
+
+|                                     | Value                                |
+| ----------------------------------- | ------------------------------------ |
+| **Rule ID**                         | CA2256                               |
+| **Category**                        | [Usage](usage-warnings.md)           |
+| **Fix is breaking or non-breaking** | Non-breaking                         |
+
+## Cause
+
+An interface with <xref:System.Runtime.InteropServices.DynamicInterfaceCastableImplementationAttribute> has a non-implemented member.
+
+## Rule description
+
+Types attributed with <xref:System.Runtime.InteropServices.DynamicInterfaceCastableImplementationAttribute> act as an interface implementation for a type that implements the 'IDynamicInterfaceCastable' type. As a result, it must provide an implementation of all of the members defined in the inherited interfaces, because the type that implements 'IDynamicInterfaceCastable' will not provide them otherwise.
```

Comment:
```suggestion
Types attributed with <xref:System.Runtime.InteropServices.DynamicInterfaceCastableImplementationAttribute> act as an interface implementation for a type that implements the `IDynamicInterfaceCastable` type. As a result, it must provide an implementation of all of the members defined in the inherited interfaces, because the type that implements `IDynamicInterfaceCastable` will not provide them otherwise.
```